WordPress/gutenberg
# Add table block caption

Related #15283, #6923

Fixes #11589

## Description
- Adds the ability to specify a caption for the table block.

## How has this been tested?
1. Add a table
2. Add a caption to the table
3. Preview the post and check that the table has a caption

## Screenshots 
Side-by-side of image and table captions in the editor
![Screen Shot 2019-05-10 at 4 00 23 pm](https://user-images.githubusercontent.com/677833/57512082-4dfaeb80-733d-11e9-8ceb-a63b36d17161.png)


## Types of changes


New feature (non-breaking change which adds functionality)


## Checklist:
